rsocket-examples/src/main/java/io/rsocket/examples/transport/tcp/plugins/LimitRateInterceptoreExample.java

```diff
@@ -0,0 +1,135 @@
+
+  public static void main(String[] args) {
+    BlockingQueue<String> requests = new ArrayBlockingQueue<>(100);
+    RSocketServer.create(
+            (sp, sr) ->
+                Mono.just(
+                    new RSocket() {
+                      @Override
+                      public Flux<Payload> requestStream(Payload payload) {
+                        return Flux.interval(Duration.ofMillis(100))
+                            .doOnRequest(e -> requests.add("Responder requestN(" + e + ")"))
+                            .map(aLong -> DefaultPayload.create("Interval: " + aLong));
+                      }
+
+                      @Override
+                      public Flux<Payload> requestChannel(Publisher<Payload> payloads) {
+                        return Flux.from(payloads)
+                            .doOnRequest(e -> requests.add("Responder requestN(" + e + ")"));
+                      }
+                    }))
+        .interceptors(
+            ir ->
+                ir.forRequester((RSocketInterceptor) r -> new LimitRateRSocket(r, true))
+                    .forResponder((RSocketInterceptor) r -> new LimitRateRSocket(r, false)))
+        .bind(TcpServerTransport.create("localhost", 7000))
+        .subscribe();
+
+    RSocket socket =
+        RSocketConnector.create()
+            .interceptors(
+                ir ->
+                    ir.forRequester((RSocketInterceptor) r -> new LimitRateRSocket(r, true))
+                        .forResponder((RSocketInterceptor) r -> new LimitRateRSocket(r, false)))
+            .connect(TcpClientTransport.create("localhost", 7000))
+            .block();
+
+    socket
+        .requestStream(DefaultPayload.create("Hello"))
+        .doOnRequest(e -> requests.add("Requester requestN(" + e + ")"))
+        .map(Payload::getDataUtf8)
+        .doOnNext(logger::debug)
+        .take(10)
+        .then()
+        .block();
+
+    requests.forEach(request -> logger.debug("Requested : {}", request));
+    requests.clear();
+
+    logger.debug("-----------------------------------------------------------------");
+    logger.debug("Does requestChannel");
+    socket
+        .requestChannel(
+            Flux.<Payload, Long>generate(
+                    () -> 1L,
+                    (s, sink) -> {
+                      sink.next(DefaultPayload.create("Next " + s));
+                      return ++s;
+                    })
+                .doOnRequest(e -> requests.add("Requester Upstream requestN(" + e + ")")))
+        .doOnRequest(e -> requests.add("Requester Downstream requestN(" + e + ")"))
+        .map(Payload::getDataUtf8)
+        .doOnNext(logger::debug)
+        .take(10)
+        .then()
+        .doFinally(signalType -> socket.dispose())
+        .then()
+        .block();
+
+    requests.forEach(request -> logger.debug("Requested : {}", request));
+  }
+
+  static class LimitRateRSocket extends RSocketProxy {
+
```

Comment:
This could look better as a `LimitRateInterceptor` in a top-level class, applying the proxy internally.

Reply:
fixed